File: 프로그래머스/lv2/12980. 점프와 순간 이동/점프와 순간 이동.js
function solution(n){
    let count = 0;
    
    while(n !== 0) {
        if(n % 2 === 0) { // 짝수
            n = n/2;
        }
        else {  // 홀수
            n--;
            count++;
        }
    }
    return count;
    
}

/*
홀수면 -1 해서 짝수로 만들고 걸음수++
짝수면 /2 해서 순간이동. 
종착지는 0이 될 때 까지 반복한다.
*/

File: 프로그래머스/2/12980. 점프와 순간 이동/점프와 순간 이동.py
import math

def solution(n):
    temp = n
    battery = 0
    
    while(True):
        if temp <= 0:
            return battery
        
        d = temp / 2
        if d == math.floor(d):
            temp = math.floor(d)
            
        else:
            temp -= 1
            battery += 1
            
            